tools/codesign_macOS_executable/sign_executable.py

```diff
@@ -0,0 +1,177 @@
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libApexFramework.dylib"),
+            os.path.join(executable, "Contents", "MacOS", os.path.splitext(executable_name)[0])]
+
+        assert os.path.exists(args.entitlements_file)
+
+        for file in sign_files:
+            cmd = [
+                "sudo", "codesign", "-f", "-s", "-v", "--options", "runtime", "--timestamp", "--entitlements", 
+                args.entitlements_file, "--sign", f"Developer ID Application: {args.developer_id}", file]
+            print(f"[SPEAR | sign_executable.py] Executing: {' '.join(cmd)}")
+            cmd_result = subprocess.run(cmd)
+            assert cmd_result.returncode == 0
+
+        os.makedirs(args.output_dir, exist_ok=True)
+        os.makedirs(args.tmp_dir, exist_ok=True)
+
+        # create a zip file for notarization
+        notrn_zip = os.path.join(args.tmp_dir, f"{os.path.splitext(os.path.basename(executable))[0]}.zip")
+        cmd = ["ditto", "-c", "-k", "--rsrc", "--keepParent", executable, notrn_zip]
+        print(f"[SPEAR | sign_executable.py] Executing: {' '.join(cmd)}")
+        cmd_result = subprocess.run(cmd)
+        assert cmd_result.returncode == 0
+
+        # send the zip file for notarization
+        cmd = [
+            "xcrun", "altool", "--notarize-app", "--primary-bundle-id", "org.embodiedaifoundation.spear", 
+            "--username", args.apple_username, "--password", args.apple_password, "--file", notrn_zip]
+        print(f"[SPEAR | sign_executable.py] Executing: {' '.join(cmd)}")
+        ps = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        output = subprocess.check_output(["grep", "RequestUUID = "], stdin=ps.stdout, text=True)
+        ps.wait()
+        ps.stdout.close()
+        request_uuid = output.split("RequestUUID = ")[1].strip()
+        print(f"[SPEAR | sign_executable.py] Zip file sent for notarization. Request UUID: {request_uuid}")
+
+    # check notarization status
+    cmd = ["xcrun", "altool", "--notarization-info", request_uuid, "--username", args.apple_username, "--password", args.apple_password]
+    print(f"[SPEAR | sign_executable.py] Executing: {' '.join(cmd)}")
+    output = "in progress"
+    start_time = time.time()
+    elapsed_time = time.time() - start_time
+    while output == "in progress" and elapsed_time < args.wait_time_seconds:
+        ps = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        output = subprocess.check_output(["grep", "     Status:"], stdin=ps.stdout, text=True)
+        output = output.split("     Status:")[1].strip()
+        ps.wait()
+        ps.stdout.close()
+        elapsed_time = time.time() - start_time
+        print(f"[SPEAR | sign_executable.py] Waiting to get more information on the notarization request uuid: {request_uuid}, current status is {output} ...")
```

Comment:
- be consistent with these print statements
- Waiting to get more information on the notarization request UUID {request_uuid}, current status is {output}...
- Exceeded maximum wait time ({args.wait_time_seconds}s) for request UUID {request_uuid}, please complete the rest of the procedure after notarization is complete."

Reply:
Done